src/main/java/org/apache/commons/collections4/bloomfilter/EnhancedDoubleHasher.java

```diff
@@ -167,41 +167,33 @@ public boolean processIndices(final IntPredicate consumer) {
                 // hash[i] = ( h1(x) - i*h2(x) - (i*i*i - i)/6 ) wrapped in [0, bits)
 
                 int index = BitMaps.mod(initial, bits);
+                if (!consumer.test(index)) {
+                    return false;
+                }
                 int inc = BitMaps.mod(increment, bits);
 
                 final int k = shape.getNumberOfHashFunctions();
-                if (k > bits) {
-                    for (int j = k; j > 0;) {
-                        // handle k > bits
-                        final int block = Math.min(j, bits);
-                        j -= block;
-                        for (int i = 1; i <= block; i++) {
-                            if (!consumer.test(index)) {
-                                return false;
-                            }
-                            // Update index and handle wrapping
-                            index -= inc;
-                            index = index < 0 ? index + bits : index;
-
-                            // Incorporate the counter into the increment to create a
-                            // tetrahedral number additional term, and handle wrapping.
-                            inc -= i;
-                            inc = inc < 0 ? inc + bits : inc;
-                        }
+
+                // the tetraheadral incrementer.  We need to ensure that this
+                // number does not exceed bits-1 or we may end up with an index > bits.
+                int tet = 1;
+                for (int i = 1; i < k; i++) {
+                    // Update index and handle wrapping
+                    index -= inc;
+                    index = index < 0 ? index + bits : index;
+                    if (!consumer.test(index)) {
+                        return false;
+                    }
+
+                    // Incorporate the counter into the increment to create a
+                    // tetrahedral number additional term, and handle wrapping.
+                    inc -= tet;
+                    inc = inc < 0 ? inc + bits : inc;
+                    if (inc >= bits) {
+                        inc = BitMaps.mod(increment, bits);
```

Comment:
Don't do this. The entire purpose of the increment being tested against zero and adjusted by `[0, bits)` is to avoid a modulus inside the loop.

Besides, you should be testing `tet` is within `[0, bits)`, not `inc`.

I feel that the better solution is to have two versions. One for the idiot who requests more hash functions than there are bits in the filter; the other one optimised for the correct use case.

Reply:
OK. But I still think this is the wrong solution. You are compromising 99.9999% of use cases to handle the edge case of `k >= bits`.

Reply:
I assume this PR is not ready then?